x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersion.java

```diff
@@ -0,0 +1,389 @@
+    public static final MlConfigVersion V_7_17_0 = registerLegacyMlConfigVersion(7_17_00_99, "056316CB-61CB-4A5D-84BB-8ACDA24D5748");
+    public static final MlConfigVersion V_7_17_1 = registerLegacyMlConfigVersion(7_17_01_99, "A0006AC5-7C5A-42CC-97F7-8792FC8DEEAC");
+    public static final MlConfigVersion V_7_17_8 = registerLegacyMlConfigVersion(7_17_08_99, "A769CAD5-0A3F-4CB4-BC0C-9CFDCC6D6605");
+    public static final MlConfigVersion V_8_0_0 = registerLegacyMlConfigVersion(8_00_00_99, "2B5AFCFD-90F9-41CD-9E55-4D2711F03758");
+    public static final MlConfigVersion V_8_1_0 = registerLegacyMlConfigVersion(8_01_00_99, "4E1B967E-4BFA-4C41-8F59-E43C87B1DEFF");
+    public static final MlConfigVersion V_8_2_0 = registerLegacyMlConfigVersion(8_02_00_99, "01E035F3-D864-4094-8523-84D717EFB89D");
+    public static final MlConfigVersion V_8_3_0 = registerLegacyMlConfigVersion(8_03_00_99, "574B950D-2E9F-4CCA-9E61-F93B4420D800");
+    public static final MlConfigVersion V_8_4_0 = registerLegacyMlConfigVersion(8_04_00_99, "D92DAC63-DAE6-4858-9E63-9BD91B67459A");
+    public static final MlConfigVersion V_8_5_0 = registerLegacyMlConfigVersion(8_05_00_99, "178A2E20-8F8E-4430-A002-B156E1E99B22");
+    public static final MlConfigVersion V_8_6_0 = registerLegacyMlConfigVersion(8_06_00_99, "C910CB2A-A7B5-47AE-A9F5-FAC8F5205D27");
+    public static final MlConfigVersion V_8_6_1 = registerLegacyMlConfigVersion(8_06_01_99, "DFD4116C-1173-4461-8A00-00E0ADCAD2C2");
+    public static final MlConfigVersion V_8_7_0 = registerLegacyMlConfigVersion(8_07_00_99, "C9A8AACB-84FA-44C6-A541-2FDEAECB280D");
+    public static final MlConfigVersion V_8_7_1 = registerLegacyMlConfigVersion(8_07_01_99, "B2EC2F2F-9D73-4057-A21C-23E0EF3AD311");
+    public static final MlConfigVersion V_8_8_0 = registerLegacyMlConfigVersion(8_08_00_99, "8E50EED5-54E3-45B1-A3B2-83A44ADBBF09");
+    public static final MlConfigVersion V_8_8_1 = registerLegacyMlConfigVersion(8_08_01_99, "99A928F3-FD13-4325-9770-317EB624C85C");
+    public static final MlConfigVersion V_8_9_0 = registerLegacyMlConfigVersion(8_09_00_99, "C50F56AB-4DB8-48A5-9467-4F5B07365C5C");
+
+    /*
+     * READ THE JAVADOC ABOVE BEFORE ADDING NEW ML CONFIG VERSIONS
+     * Detached ML config versions added below here.
+     */
+
+    public static final MlConfigVersion V_10 = registerMlConfigVersion(10, "4B940FD9-BEDD-4589-8E08-02D9B480B22D");
+
+    /**
+     * Reference to the most recent Ml config version.
+     * This should be the Ml config version with the highest id.
+     */
+    public static final MlConfigVersion CURRENT = V_10;
+
+    /**
+     * Reference to the most recent legacy Ml config version.
+     * This is the highest id associated with Ml config versions
+     * that use the legacy, semantic style, representation of the version.
+     */
+    private static final MlConfigVersion MAXIMUM_LEGACY_VERSION = V_8_9_0;
```

Comment:
We're not going to merge this change to 8.9.1, so it would be better to make this constant 8.10.0 and name it `CUTOVER_VERSION` or something like that. Then the tests based on it should be `before` rather than `onOrBefore`.